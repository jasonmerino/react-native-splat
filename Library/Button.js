import React, {
  Component,
  PropTypes,
  StyleSheet,
  Text,
  TouchableHighlight,
  ActivityIndicatorIOS,
  View,
} from 'react-native';

const componentStyles = StyleSheet.create({
  button: {
    height: 48,
    backgroundColor: '#333',
    justifyContent: 'center',
    borderRadius: 3,
  },
  textStyle: {
    color: 'white',
    textAlign: 'center',
  },
});

class Button extends Component {


  getContent() {
    const { children, isLoading } = this.props;
    if (isLoading) {
      return (
        <View>
          <ActivityIndicatorIOS />
        </View>
      );
    }
    if (typeof children === 'string') {
      return <Text style={[componentStyles.textStyle, this.props.textStyle]}>{children}</Text>;
    }
    return this.props.children;
  }

  render() {
    return (
      <TouchableHighlight
        onPress={this.props.onPress}
        style={[componentStyles.button, this.props.style]}
        underlayColor={this.props.underlayColor}
      >
        {this.getContent()}
      </TouchableHighlight>
    );
  }

}

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string,
  ]),
  style: Text.propTypes.style,
  textStyle: Text.propTypes.style,
  onPress: PropTypes.func,
  isLoading: PropTypes.bool.isRequired,
  underlayColor: PropTypes.string,
};

Button.defaultProps = {
  style: {},
  textStyle: {},
  isLoading: false,
  underlayColor: '#222',
};

export default Button;
