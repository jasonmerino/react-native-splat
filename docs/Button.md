`Button` (component)
====================



Props
-----

### `children`

type: `node`


### `disabled`

type: `bool`
defaultValue: `false`


### `isLoading`

type: `bool`
defaultValue: `false`


### `onPress`

type: `func`


### `onPressDisabled`

type: `func`


### `style`

type: `custom`
defaultValue: `{}`


### `textStyle`

type: `custom`
defaultValue: `{}`


### `underlayColor`

type: `string`
defaultValue: `'#222'`

