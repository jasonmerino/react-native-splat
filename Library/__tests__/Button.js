import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../Button';

it('renders correctly', () => {
  const tree = renderer.create(
    <Button>Button Test</Button>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
