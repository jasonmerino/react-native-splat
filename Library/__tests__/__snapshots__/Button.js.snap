exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "backgroundColor": "#333",
          "borderRadius": 3,
          "height": 48,
          "justifyContent": "center",
        },
        Object {},
      ],
    ]
  }
  testID={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "white",
          "textAlign": "center",
        },
        Object {},
      ]
    }>
    Button Test
  </Text>
</View>
`;
